reactivesocket-transport-tcp/src/main/java/io/reactivesocket/transport/tcp/client/TcpReactiveSocketConnector.java

```diff
@@ -1,81 +1,50 @@
-/**
- * Copyright 2016 Netflix, Inc.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
 package io.reactivesocket.transport.tcp.client;
 
-import io.netty.channel.EventLoopGroup;
-import io.reactivesocket.*;
-import io.reactivesocket.internal.rx.EmptySubscription;
-import io.reactivesocket.rx.Completable;
+import io.netty.buffer.ByteBuf;
+import io.reactivesocket.ConnectionSetupPayload;
+import io.reactivesocket.ReactiveSocket;
+import io.reactivesocket.ReactiveSocketConnector;
+import io.reactivex.netty.protocol.tcp.client.TcpClient;
 import org.reactivestreams.Publisher;
-import org.reactivestreams.Subscriber;
-import org.reactivestreams.Subscription;
 
 import java.net.SocketAddress;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
 import java.util.function.Consumer;
+import java.util.function.Function;
 
-/**
- * An implementation of {@link ReactiveSocketConnector} that creates Netty TCP ReactiveSockets.
- */
 public class TcpReactiveSocketConnector implements ReactiveSocketConnector<SocketAddress> {
-    private final ConnectionSetupPayload connectionSetupPayload;
+
+    private final ConcurrentMap<SocketAddress, TcpReactiveSocketFactory> socketFactories;
```

Comment:
API-wise I think it's more user-friendly to ask for:
- `Publisher<List<SocketAddress>>` as the source
- A `ReactiveSocketConnector` as the code responsible for transforming the address into a `ReactiveSocket`

Rather than `Publisher<List<ReactiveSocketFactory>>`.

It allows separation of concern between something like the discovery mechanism (Eureka, DNS...) and the Transport (TCP, WebSocket, Aeron).

You're right than the `ReactiveSocketFactory` provides barely more abstraction than a raw `ReactiveSocketConnector` (it only hides the SocketAddress). But I believe it makes the API cleaner.